hooks/useDetectOs.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect, useState } from 'react';
+import { detectOS } from '../util/detectOS';
+import { downloadUrlByOS } from '../util/downloadUrlByOS';
+
+import type { UserOS } from '../types/userOS';
+
+export const useDetectOS = () => {
+  const [userOS, setUserOS] = useState<UserOS>('UNKNOWN');
+  const [bitness, setBitness] = useState('');
+
+  useEffect(() => {
+    setUserOS(detectOS);
```

Comment:
Shouldn't this be a function call? Like `setUserOS(detectOS())`?

Reply:
I don't think it's required, but it indeed looks better and clearer with `setUserOS(detectOS())`